--- a/libgcc/config/i386/linux-unwind.h	2018-06-29 14:39:11.371628174 +0200
+++ b/libgcc/config/i386/linux-unwind.h	2018-06-29 14:42:52.161628174 +0200
@@ -58,7 +58,7 @@
   if (*(unsigned char *)(pc+0) == 0x48
       && *(unsigned long long *)(pc+1) == RT_SIGRETURN_SYSCALL)
     {
-      struct ucontext *uc_ = context->cfa;
+      struct ucontext_t *uc_ = context->cfa;
       /* The void * cast is necessary to avoid an aliasing warning.
          The aliasing warning is correct, but should not be a problem
          because it does not alias anything.  */
